db/memtable.cc

```diff
@@ -1011,4 +1012,21 @@ uint64_t MemTable::GetMinLogContainingPrepSection() {
   return min_prep_log_referenced_.load();
 }
 
+bool MemTable::ShouldFlushAfterSampledRead() {
+  // TODO: Make the two constants options.
+  static const int kSampleRate = 1024;
+  static const int kNumReadsUntilFlush = 1024 * 1024;
```

Comment:
This number feels far too small to me for triggering a flush. RocksDB can do many millions of Get() per second. We don't want to trigger several flushes per second.
And it may be a good idea to pro-rated based on memtable size.

LevelDB has comment describing how they came up with their triggering condition: https://github.com/google/leveldb/blob/master/db/version_set.cc#L671-L683

I think we need some thinking about the triggering condition, and I suggest we are conservative in the beginning.

Reply:
Idea was to make these options that are by default off, the current constants are just for initial testing (see description). But yeah I can look more into having a different triggering condition.